src/MiNET/MiNET/Utils/Skins/Skin.cs

```diff
@@ -126,5 +127,18 @@ public static string ToJson(GeometryModel geometryModel)
 
 			return JsonConvert.SerializeObject(geometryModel, settings);
 		}
+
+		public object Clone()
+		{
+			return new Skin
+			{
+				SkinId = SkinId,
+				SkinGeometry = SkinGeometry,
+				SkinGeometryName = SkinGeometryName,
+				Slim = Slim,
+				SkinData = SkinData?.ToArray(),
```

Comment:
Please check PaletteChunk Clone() for how to clone the arrays.
```suggestion
				SkinData?.CopyTo(...., 0),
```
Linq is hopelessly slow and sluggish.

Reply:
Changed to Array.CopyTo() in ea83bb4